ExampleMod/Content/Items/ExampleResearchPresent.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.GameContent.Creative;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Items
+{
+	public class ExampleResearchPresent : ModItem
+	{
+
+		public override void SetStaticDefaults() {
+			DisplayName.SetDefault("Example Present");
+			Tooltip.SetDefault("Contains a random accessory! Try researching it for infinite of it!");
+			CreativeItemSacrificesCatalog.Instance.SacrificeCountNeededByItemId[Type] = ItemLoader.ItemCount;
```

Comment:
Does sacrifice count changing when you add/remove mods cause a problem? What if you sacrifice 5k of a thing and then unload a mod bringing the count needed down to 4k. Does the item become fully researched then and then become unresearched when you add more mods?

Reply:
I changed the Research value from a random integer (1-100) to ItemCount and back, and if the player already researched fully the present, would still be able to research it further because of CanResearch returning always true. Also, when changing the values, the item went from fully researched to only partially researched whenever it went from lower to higher numbers, but researching one new item will reduce the amount researched to the maximum at the time.